Tools/autotest/arducopter.py

```diff
@@ -6183,6 +6183,77 @@ def DynamicRpmNotches(self):
         if ex is not None:
             raise ex
 
+    def DynamicRpmNotchesRateThread(self):
+        """Use dynamic harmonic notch to control motor noise via ESC telemetry."""
+        self.progress("Flying with ESC telemetry driven dynamic notches")
+
+        self.set_rc_default()
+        self.set_parameters({
+            "AHRS_EKF_TYPE": 10,
+            "INS_LOG_BAT_MASK": 3,
+            "INS_LOG_BAT_OPT": 0,
+            "INS_GYRO_FILTER": 300, # set gyro filter high so we can observe behaviour
+            "LOG_BITMASK": 959,
+            "LOG_DISARMED": 0,
+            "SIM_VIB_MOT_MAX": 350,
+            "SIM_GYR1_RND": 20,
+            "SIM_ESC_TELEM": 1,
+            "FLIGHT_OPTIONS": 8
+        })
+        self.reboot_sitl()
```

Comment:
We need a `context_push()` at the top of this test and `context_pop()` at the bottom followed by the existing `reboot_sitl()`

... otherwise subsequent tests will end up running with the wrong parameters in effect (not value, effect).  Assuming this reboot_sitl() is here for a reason....

Reply:
Fixed